Preprocessing/oxygenPP.m

```diff
@@ -0,0 +1,214 @@
+if strcmpi(qcLevel, 'raw'), return; end
+
+% auto logical in input to enable running under batch processing
+if nargin<3, auto=false; end
+
+for k = 1:length(sample_data)
+  
+  sam = sample_data{k};
+  
+  % Get the temperature
+  tempIdx       = getVar(sam.variables, 'TEMP');
+  if ~(tempIdx), continue; end % if there is no temperature, then don't do anythin  
+  % get coorinates and dimensions from TEMP to use as output
+  % coordinates/dimensions
+  dimensions = sam.variables{tempIdx}.dimensions;
+  if isfield(sam.variables{tempIdx}, 'coordinates')
+      coordinates = sam.variables{tempIdx}.coordinates;
+  else
+      coordinates = '';
+  end
+  temp = sam.variables{tempIdx}.data;
+  
+  % indexs of other variables that might be needed
+  psalIdx       = getVar(sam.variables, 'PSAL');
+  
+  [presRel, presName] = presRelFromSampleData(sam);
+
+	% from imosParameters.txt
+	% DOX,                 0, volume_concentration_of_dissolved_molecular_oxygen_in_sea_water,                          ml l-1,        ,              ,                                  O, 999999.0, 0.0,      200.0,    float
+	% DOX1,                1, mole_concentration_of_dissolved_molecular_oxygen_in_sea_water,                            umol l-1,      ,              ,                                  O, 999999.0, 0.0,      1000.0, float
+	% DOX2,                1, moles_of_oxygen_per_unit_mass_in_sea_water,                                               umol kg-1,     ,              ,                                  O, 999999.0, 0.0,      1000.0, float
+	% DOXS,                1, fractional_saturation_of_oxygen_in_sea_water,                                             percent,       ,              ,                                  O, 999999.0, ,         ,         float
+	% DOXY,                1, mass_concentration_of_oxygen_in_sea_water,                                                kg m-3,        ,              ,                                  O, 999999.0, 0.0,      29.0,     float
+	% DOXY_TEMP,           1, temperature_of_sensor_for_oxygen_in_sea_water,                                            degrees_Celsius,,             ,                                  T, 999999.0, 0.0,      50.0,     float
+
+  % oxygen parameters that maybe in the dataset
+  doxIdx        = getVar(sam.variables, 'DOX');
+  dox1Idx       = getVar(sam.variables, 'DOX1');
+  dox2Idx       = getVar(sam.variables, 'DOX2');
+  doxsIdx       = getVar(sam.variables, 'DOXS');
+  doxyIdx       = getVar(sam.variables, 'DOXY');
+  
+  % check is psal and some dox measurement present, if not skip
+  if ~(psalIdx && (doxIdx || dox1Idx || dox2Idx || doxsIdx)), continue; end
+    
+  psal = sam.variables{psalIdx}.data;
+  
+  ptmp = sw_ptmp(psal, temp , presRel, 0);
+  
+  % calculate the density at 0 dbar
+  dens = sw_dens0(psal, ptmp);
+
+  % calculate oxygen solability
+  oxsol = gsw_O2sol_SP_pt(psal, ptmp); % sea bird calculates this using psal, temp, not potential temperature
```

Comment:
How about we change this to the EOS-80 function then? Oh its in ml/l not Gordon and Garcia re-fit, I have see that code somewhere, oh its already under Util/O2sol.m maybe Ill change that then